lib/r10k/git/rugged/credentials.rb

```diff
@@ -125,4 +143,63 @@ def get_git_username(url, username_from_url)
 
     user
   end
+
+  def github_app_token(app_id, private_key, ttl)
+    raise R10K::Git::GitError, _('App id contains invalid characters.') unless app_id =~ /^\d+$/
+    raise R10K::Git::GitError, _('Token ttl contains invalid characters.') unless ttl =~ /^\d+$/
+    raise R10K::Git::GitError, _('App key is missing or unreadable') unless File.readable?(private_key)
+
+    begin
+      ssl_key = OpenSSL::PKey::RSA.new(File.read(private_key).strip)
+      unless ssl_key.private?
+        raise R10K::Git::GitError, _('App key is not a valid SSL private key')
+      end
+    rescue OpenSSL::PKey::RSAError
+      raise R10K::Git::GitError, _('App key is not a valid SSL key')
+    end
+
+    logger.debug2 _("Using Github App id %{app_id} with token from %{token_path}") % { token_path: private_key, app_id: app_id }
```

Comment:
We haven't gotten a token at this point yet, right? We're using the key from the `private_key` path to generate a token below, yes?
```suggestion
    logger.debug2 _("Using Github App id %{app_id} with SSL key from %{key_path}") % { key_path: private_key, app_id: app_id }
```